src/Psy/CodeCleaner.php

```diff
@@ -53,7 +52,9 @@ class CodeCleaner
      */
     public function __construct(Parser $parser = null, Printer $printer = null, NodeTraverser $traverser = null)
     {
-        $this->parser    = $parser    ?: new Parser(new Lexer());
+        $parserFactory = new ParserFactory();
```

Comment:
Agree, fixed to `if`. Hope we will see PHP version increment in the requirements soon...